code/workhorse/screen_index_helper.R

```diff
@@ -0,0 +1,142 @@
+compute_covariance_h <- function(simulator,set1,set2){
+  #set2 can be pool or benchmark or the merge of these two
+  # set1= as.matrix(set1)
+  # set2 = as.matrix(set2)
+  
+  d = simulator$model[["numDim"]]
+  beta = simulator$model[["beta"]]
+  sigma = simulator$model[["sig2"]]
+  number = nrow(set1)
+  newcov=matrix(nrow=dim(set2)[1],ncol=number)
+  
+  for ( i in 1:dim(set2)[1]) {
+    for (j in 1:number) {
+      dist= -sum((set2[i,1:d]-set1[j,1:d])^2*beta)
+      newcov[i,j]= sigma*exp(dist)
+    }
+  }
+  cat("compute kernel done \n")
+  return(newcov)
+}
+
+# you have three metrics, so you should have three functions to compute the metrics.
+# This is the outline
+predict_testset <- function(simulator,testset, unfold=FALSE) {
+  #unfold is true, return the mean and covariance.
+  sigma = simulator$model[["sig2"]]
+  inverse=simulator$model$invVarMatrix
+  mu=simulator$model[["mu"]]
+  fitZ = simulator$model$Z
+  nugget= simulator$model[["nugget"]]
+  
+  newcov = compute_covariance_h(simulator,set1=simulator$train,set2 = testset)
+  
+  diag(newcov)=diag(newcov) #+nugget
+  mean=newcov%*%inverse%*%(fitZ - mu)+mu[1]
+  var=diag(sigma+nugget-newcov%*%inverse%*%t(newcov))
+  mean_var = list("mean"=mean,"var"=var)
+  
+  real = testset[,ncol(testset)]
+  MAE = mean(abs(mean-real))
+  
+  return(if (unfold) mean_var else MAE)
+}
+
+EN_metric <- function(simulator){
+  res =  predict_testset(simulator,simulator$pool[, 1:14], unfold=TRUE) 
+  var = res$var
+  return(var)
+}
+
+compute_MI_denominator <- function(simulator){
+  
+  d = simulator$model[["numDim"]]
+  beta = simulator$model[["beta"]]
+  sigma = simulator$model[["sig2"]]
+  nugget=simulator$model[["nugget"]]
+  
+  ## cov(unob, unob)
+  cov_unob = compute_covariance_h(simulator,simulator$pool[, 1:14], simulator$pool[, 1:14])
```

Comment:
Sse `<-` for assignment everywhere except inside lists and function calls. I now see this is applicable to many places in the code.

Reply:
updated.